src/elisp/treemacs-mouse-interface.el

```diff
@@ -37,6 +37,16 @@
   (require 'cl-lib)
   (require 'treemacs-macros))
 
+(defgroup treemacs-mouse nil
+  "Treemacs+mouse configuration options."
+  :group 'treemacs-mouse
+  :prefix "treemacs-mouse-")
+
+(defcustom treemacs-mouse-move-files-by-dragging t
+  "If non-nil, move files by dragging with your mouse inside treemacs."
+  :group 'treemacs-mouse
```

Comment:
The new variable needs to be advertised in 2 places in the readme: in the install use-package example and the table of all configuration variables. There should also be a line in the Changelog.org.